xarray/tests/__init__.py

```diff
@@ -114,6 +114,18 @@ def requires_bottleneck(test):
     return test if has_bottleneck else pytest.mark.skip('requires bottleneck')(test)
 
 
+flakey = pytest.mark.skipif(
+    pytest.config.getoption("--skip-flakey"),
+    reason="set --skip-flakey option to skip flakey tests"
+)
+
+
+slow = pytest.mark.skipif(
+    not pytest.config.getoption("--run-slow"),
+    reason="set --run-slow option to run slow tests"
```

Comment:
I've set the three categories:
 * slow (runs on default): `slow`
 * CI specific (runs on default, specified to be optional for CI): `optionalci`
 * flakey (not run on default): `flakey`

I don't currently have flakey tests.  Plotting routines are marked as slow and long tests from the too many open files issue are marked `optionalci`.  We have some test (h5netcdf) that are flakey because of that bug I've found.  They are no longer commented but are flagged via `@flakey`.